src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -694,8 +701,8 @@ public Future<?> onError(Throwable e)
             state(State.FAILED); // make sure subsequent error handling sees the session in a final state 
             channel.sendControlMessage(new SessionFailedMessage()).awaitUninterruptibly();
         }
-
-        return closeSession(State.FAILED);
+        String exception = Throwables.getStackTraceAsString(e);
+        return closeSession(State.FAILED, "Failed because of an unkown exception\n" + exception.substring(0,exception.length() -143)); //bound the exception by removing internal line logs
```

Comment:
I don't feel this address my feedback... I was asking that you define a solution to bound memory that you could justify; can you justify why this solution?